Sources/Networking/Core/DownloadAPIManaging.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  DownloadAPIManaging.swift
+//  
+//
+//  Created by Dominika Gajdová on 12.05.2023.
+//
+
+import Foundation
+
+// MARK: - Defines Download API managing
+public typealias DownloadResult = (URLSessionDownloadTask, Response)
+
+/// A definition of an API layer with methods for handling data downloading.
+/// Recommended to be used as singleton.
+/// If you wish to use multiple instances, make sure you manually invalidate url session by calling the `invalidateSession` method.
+public protocol DownloadAPIManaging {
+    /// List of all currently ongoing download tasks.
+    var allTasks: [URLSessionDownloadTask] { get async }
+    
+    /// Invalidates urlSession instance.
+    /// - Parameters:
+    ///   - shouldFinishTasks: Indicates whether all currently active tasks should be able to finish before invalidating. Otherwise they will be cancelled.
+    func invalidateSession(_ shouldFinishTasks: Bool)
```

Comment:
```suggestion
    func invalidateSession(shouldFinishTasks: Bool)
```

Otherwise `invalidateSession(false)` could be misleading.